src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -71,8 +71,10 @@
   private NavigableMap<Long, HttpRequestMetricsSnapshot> httpRqMetricsMap;
   private NavigableMap<Long, MasterEventMetricsSnapshot> masterEventMetricsMap;
   private Map<AllMetrics.MetricName, NavigableMap<Long, MemoryDBSnapshot>> nodeMetricsMap;
+  private NavigableMap<Long, ShardStateMetricsSnapshot> shardStateMetricsMap;
   private static final int MAX_DATABASES = 2;
   private static final int OS_SNAPSHOTS = 4;
+  private static final int SHARD_STATE_SNAPSHOTS = 4;
```

Comment:
We will consume a lot of bytes on heap if we keep 4 snapshots around. Maybe 2 is good enough.

Reply:
Changed the value to 2